osf/migrations/0121_migrate_registrations_to_osf_registries_provider.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.13 on 2018-07-18 13:51
+from __future__ import unicode_literals
+import logging
+
+from django.db import migrations
+from django.db.models import Q
+from django.contrib.contenttypes.models import ContentType
+
+logger = logging.getLogger(__file__)
+
+
+def migrate_existing_registrations_into_osf_registries(state, schema):
+    RegistrationProvider = state.get_model('osf', 'registrationprovider')
+    Registration = state.get_model('osf', 'registration')
+    Guid = state.get_model('osf', 'Guid')
+    osf_registries = RegistrationProvider.objects.get(name='OSF Registries')
+
+    regs = Registration.objects.filter(
+        is_public=True,
+        is_deleted=False,
+    ).filter(
+        Q(Q(embargo__isnull=True) | ~Q(embargo__state='unapproved')) &
+        Q(Q(retraction__isnull=True) | ~Q(retraction__state='approved'))
+    )
+
+    for reg in regs:
```

Comment:
This will take a very long time. I started it locally about 45 minutes ago and it's still not done. Consider doing something like this instead:
```python
reg_infos = list(Registration.objects.filter([query]).values('id', 'creator_id'))
contenttype_id = ContentType.objects.get(model='abstractnode').id
cgm_to_create = []

for reg_info in reg_ids:
    cgm_to_create.append(
        CollectedGuidMetadata(
            provider=osf_registries,
            guid=Guid.objects.get(content_type_id=contenttype_id, object_id=reg_info['id']),
            creator_id=reg_info['creator_id']
        )
    )

CollectedGuidMetadata.objects.bulk_create(cgm_to_create)
```
You may also want to add pagination (i.e. every 1000 CGMs, run `bulk_create`). There are a couple other data migrations that do something similar, you might want to give those a look.

For local optimization testing, I'd suggest making about 10k registrations in the shell before running it.

Reply:
Additionally, these numbers look off:
```
(osf) ~/D/osf.io ❯ time python manage.py migrate 
[...]
INFO: Successfully migrated 218666 public registrations into the OSF Registries.
 OK
python manage.py migrate  1403.69s user 93.33s system 17% cpu 2:25:58.78 total
```
```
In [4]: Registration.objects.filter(
   ...:         is_public=True,
   ...:         is_deleted=False,
   ...:     ).filter(
   ...:         Q(Q(embargo__isnull=True) | ~Q(embargo__state='unapproved')) &
   ...:         Q(Q(retraction__isnull=True) | ~Q(retraction__state='approved'))
   ...:     ).count()
Out[4]: 19780

In [5]: AbstractNode.objects.filter(
   ...:         is_public=True,
   ...:         is_deleted=False,
   ...:     ).filter(
   ...:         Q(Q(embargo__isnull=True) | ~Q(embargo__state='unapproved')) &
   ...:         Q(Q(retraction__isnull=True) | ~Q(retraction__state='approved'))
   ...:     ).count()
Out[5]: 218666
```
This is likely due to `TypedModel` behavior being affected by the [historical models](https://docs.djangoproject.com/en/1.11/topics/migrations/#historical-models) caveat for Django migrations. Prefer using `type='osf.registration'` in the query instead.